### PR TITLE
Add license headers to some files.

### PR DESCRIPTION
**Pull request by mosuem**
---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/intl/lib/src/global_state.dart [hunk 1]
```diff
@@ -1,3 +1,7 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
```
Comment (devoncarew) [inline pkgs/intl/lib/src/global_state.dart:1]:
```suggestion
// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
```

File: pkgs/intl/test/number_format_compact_icu_test.dart [hunk 2]
```diff
@@ -1,3 +1,7 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
```
Comment (devoncarew) [inline pkgs/intl/test/number_format_compact_icu_test.dart:1]:
```suggestion
// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
```

File: pkgs/intl/test/number_format_compact_web_test.dart [hunk 3]
```diff
@@ -1,3 +1,7 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
```
Comment (devoncarew) [inline pkgs/intl/test/number_format_compact_web_test.dart:1]:
```suggestion
// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
```

**Comment by lrhn** [comment]
Fix PR title before landing.

**Review by lrhn — APPROVED** [review]

**Pull request merged**